Dockerfile

```diff
@@ -28,32 +18,30 @@ LABEL summary="$SUMMARY" \
   io.openshift.tags="sonarqube" \
   release="$SONAR_VERSION"
 
-USER root
-EXPOSE 9000
-ADD root /
+# Define Plug-in Versions
+ARG SONAR_ZAP_PLUGIN_VERSION=1.2.0
+ENV SONARQUBE_PLUGIN_DIR="$SONARQUBE_HOME/extensions/plugins"
 
-RUN set -x \
-  && cd /opt \
-  && curl -o sonarqube.zip -fSL https://binaries.sonarsource.com/Distribution/sonarqube/sonarqube-$SONAR_VERSION.zip \
-  && unzip sonarqube.zip \
-  && mv sonarqube-$SONAR_VERSION sonarqube \
-  && rm sonarqube.zip* \
-  && rm -rf $SONARQUBE_HOME/bin/*
+# Switch to root for package installs
+USER 0
 
 # ================================================================================================================================================================================
 # Bundle Plug-in(s)
 # --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 
-# sonar-zap-plugin
-# https://github.com/Coveros/zap-sonar-plugin
-ADD https://github.com/Coveros/zap-sonar-plugin/releases/download/sonar-zap-plugin-$SONAR_ZAP_PLUGIN_VERSION/sonar-zap-plugin-$SONAR_ZAP_PLUGIN_VERSION.jar $SONARQUBE_PLUGIN_DIR
+# sonar-zap-plugin - https://github.com/Coveros/zap-sonar-plugin
+RUN set -x \
+  && cd "$SONARQUBE_PLUGIN_DIR" \
+  && curl -o "sonar-zap-plugin-$SONAR_ZAP_PLUGIN_VERSION.jar" -fsSL "https://github.com/Coveros/zap-sonar-plugin/releases/download/sonar-zap-plugin-$SONAR_ZAP_PLUGIN_VERSION/sonar-zap-plugin-$SONAR_ZAP_PLUGIN_VERSION.jar"
 
```

Comment:
I used `curl` instead of docker ADD because using ADD with a remote URL is discouraged. See:
- https://dev.to/lirantal/why-you-should-use-copy-instead-of-add-when-building-docker-images-5fkj
- https://snyk.io/blog/10-docker-image-security-best-practices/